synapse/handlers/space_summary.py

```diff
@@ -533,7 +533,7 @@ async def _build_room_entry(self, room_id: str) -> JsonDict:
             "guest_can_join": stats["guest_access"] == "can_join",
             "creation_ts": create_event.origin_server_ts,
             "room_type": create_event.content.get(EventContentFields.ROOM_TYPE),
-            "allowed_spaces": allowed_spaces,
+            "allowed_spaces": allowed_rooms,
```

Comment:
I should have mentioned that I left this at `allowed_spaces` for now since the form of this API is somewhat unrelated to the restricted rooms MSC.